### PR TITLE
Fix potential bugs

### PR DESCRIPTION
**Pull request by 6543**
potintial bugs ...
* check if err first before use var
* use engine if is param
* remove always true statement

### CODE REVIEW

**Review by zeripath — APPROVED** [review]

**Review by zeripath — APPROVED** [review]

File: modules/markup/common/linkify.go [hunk 1]
```diff
@@ -108,7 +108,7 @@ func (s *linkifyParser) Parse(parent ast.Node, block text.Reader, pc parser.Cont
 		}
 		at := bytes.IndexByte(line, '@')
 		m = []int{0, stop, at, stop - 1}
-		if m == nil || bytes.IndexByte(line[m[2]:m[3]], '.') < 0 {
+		if bytes.IndexByte(line[m[2]:m[3]], '.') < 0 {
```
Comment (6543) [inline modules/markup/common/linkify.go:111]: alredy checked in line 100 ...

**Review by lafriks — APPROVED** [review]

**Review by guillep2k — APPROVED** [review]
Nice ones!

**Comment by lunny** [comment]
@6543 could you send a backport to v1.11?

**Comment by 6543** [comment]
backport: #10518

**Pull request merged**